### PR TITLE
Reset widget global function

### LINKED ISSUE #2335: [Feature] `clearForm` action
## Summary

A way to clear a form from outside the context of the form.

## Motivation

In some use-cases, developers would like to have the form clear its fields from an independent trigger.

### PR DESCRIPTION
**Pull request by ofpiyush**
## Description
Adds `resetWidget` feature.

Fixes #2335

## Type of change

- New feature (non-breaking change which adds functionality)

## How Has This Been Tested?

> Please describe the tests that you ran to verify your changes. Provide instructions, so we can reproduce. 
> Please also list any relevant details for your test configuration.

- Test A
- Test B

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: app/client/src/components/editorComponents/actioncreator/ActionCreator.tsx [hunk 1]
```diff
@@ -1068,6 +1126,18 @@ function useApiOptionTree() {
   return apiOptionTree;
 }
 
+function useWidgetOptionTree() {
+  const widgets = useSelector(getWidgets) || {};
+  return Object.values(widgets)
+    .filter((w) => w.type !== "CANVAS_WIDGET" && w.type !== "BUTTON_WIDGET")
```
Comment (ofpiyush) [inline app/client/src/components/editorComponents/actioncreator/ActionCreator.tsx:1132]: @hetunandu suggest better solve here.

**Comment by mohanarpit** [comment]
@ofpiyush What do you think of refactoring the function name to: `setWidgetProperties(widgetName: string, options: {})`? The options object could then be:
```
options = {
  resetToDefault: boolean (default false),
  fields: {
    <fieldName>: <fieldValue>,
    <fieldName>: <fieldValue>
  }
}
``` 

This would ensure the extendability of a function like this. Thoughts?

**Comment by ofpiyush** [comment]
@mohanarpit So long as we do not try to do it with the property pane UI, I can make it work like that.

Code around the property pane assumes values to not be nested.

@hetunandu will probably enable that with his work on #3132

**Comment by hetunandu** [comment]
> @ofpiyush What do you think of refactoring the function name to: `setWidgetProperties(widgetName: string, options: {})`? The options object could then be:
> 
> ```
> options = {
>   resetToDefault: boolean (default false),
>   fields: {
>     <fieldName>: <fieldValue>,
>     <fieldName>: <fieldValue>
>   }
> }
> ```
> 
> This would ensure the extendability of a function like this. Thoughts?

@mohanarpit this would not work quite well. 
- `set` makes me feel one can set any/all property fields of the widget and it's children which is not the case as only ephemeral "Meta Properties" are supported to be set via a trigger
- this signature of specifying fields also looks cumbersome. I would rather make it a blacklist `exceptFields`
- overall i feel `setProperty` could be a different function from `resetWidget` that could have this signature without the reset option

**Review by hetunandu — APPROVED** [review]

**Pull request merged**